oxide-auth-actix/src/operations.rs

```diff
@@ -0,0 +1,78 @@
+use crate::{OAuthRequest, OAuthResponse, OxideOperation, WebError};
+use oxide_auth::{
+    endpoint::{AccessTokenFlow, AuthorizationFlow, Endpoint, RefreshFlow, ResourceFlow},
+    primitives::grant::Grant,
+};
+
+/// Authorization-related operations
+pub struct Authorize(pub OAuthRequest);
```

Comment:
These could be polymorphic:

`struct Authorize<Req = OAuthRequest>(pub Req)` and then conditionally implement `OxideOperation` when the type, the error type and response type fulfill the required trait bounds from `OxideOperation`.

Note that, in production, it is useful to wrap a `OAuthRequest` into a custom `impl WebRequest` to enrich it with additional information from the original http request that is available in the solicitor for example.

Reply:
Also, adding generic parameters need not be the correct solution. It's just something that isn't quite clear to me.

Reply:
I've updated the Message type to carry a generic "extras" parameter which can be used in implementations to send data long with the OAuthRequest or ignored entirely for simple cases.

Reply:
@asonix Resolving this since the `Extra` parameter on the message alone seems clearer!